File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/nullable/KNullableDatePropertyName.java
package com.nagopy.android.kotlinames.property.nullable;

import com.nagopy.android.kotlinames.KNullablePropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

import java.util.Date;

public class KNullableDatePropertyName implements KNullablePropertyName<Date>, KSortablePropertyName<Date> {

    private final String name;

    public KNullableDatePropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/nullable/KNullableDatePropertyName.kt
package com.nagopy.android.kotlinames.property.nullable

import com.nagopy.android.kotlinames.KNullablePropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName
import java.util.*

class KNullableDatePropertyName(val name: String) : KNullablePropertyName<Date>, KSortablePropertyName<Date> {

    override fun name(): String = name

}
